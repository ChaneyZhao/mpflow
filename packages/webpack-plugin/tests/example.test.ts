import expectToMatchExampleDist from './helpers/expectToMatchExampleDist'
import runExample from './helpers/runExample'

describe('webpack-plugin examples', () => {
  test('app', async () => {
    const result = await runExample('app')
    expect(result.errors).toMatchSnapshot('errors')
    expect(result.warnings).toMatchSnapshot('warnings')
    await expectToMatchExampleDist(result.assets, 'app')
  })

  test('image', async () => {
    const result = await runExample('image')
    expect(result.errors).toMatchSnapshot('errors')
    expect(result.warnings).toMatchSnapshot('warnings')
    await expectToMatchExampleDist(result.assets, 'image')
  })

  test('plugin', async () => {
    const result = await runExample('plugin')
    expect(result.errors).toMatchSnapshot('errors')
    expect(result.warnings).toMatchSnapshot('warnings')
    await expectToMatchExampleDist(result.assets, 'plugin')
  })

  test('sourceMap', async () => {
    const result = await runExample('sourceMap')
    expect(result.errors).toMatchSnapshot('errors')
    expect(result.warnings).toMatchSnapshot('warnings')
    await expectToMatchExampleDist(result.assets, 'sourceMap')
  })

  test('splitChunk', async () => {
    const result = await runExample('splitChunk')
    expect(result.errors).toMatchSnapshot('errors')
    expect(result.warnings).toMatchSnapshot('warnings')
    await expectToMatchExampleDist(result.assets, 'splitChunk')
  })
})
