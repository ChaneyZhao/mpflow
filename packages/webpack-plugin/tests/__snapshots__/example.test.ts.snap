// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack-plugin examples app: errors 1`] = `Array []`;

exports[`webpack-plugin examples app: warnings 1`] = `Array []`;

exports[`webpack-plugin examples plugin: errors 1`] = `Array []`;

exports[`webpack-plugin examples plugin: warnings 1`] = `Array []`;

exports[`webpack-plugin examples sourceMap: errors 1`] = `Array []`;

exports[`webpack-plugin examples sourceMap: warnings 1`] = `Array []`;

exports[`webpack-plugin examples splitChunk: errors 1`] = `Array []`;

exports[`webpack-plugin examples splitChunk: warnings 1`] = `Array []`;
